packages/ace-mode/index.js

```diff
@@ -269,942 +26,89 @@ ace.define(
     'exports',
     'module',
     'ace/lib/oop',
-    'ace/mode/doc_comment_highlight_rules',
-    'ace/mode/text_highlight_rules'
+    'ace/mode/javascript_highlight_rules'
   ],
-  function(acequire, exports, module) {
-    'use strict';
-
-    var oop = acequire('../lib/oop');
-    var DocCommentHighlightRules = acequire('./doc_comment_highlight_rules')
-      .DocCommentHighlightRules;
-    var TextHighlightRules = acequire('./text_highlight_rules')
-      .TextHighlightRules;
-    var identifierRe = '[a-zA-Z\\$_\u00a1-\uffff][a-zA-Z\\d\\$_\u00a1-\uffff]*';
-
-    var MongoDBHighlightRules = function(options) {
-      var keywordMapper = this.createKeywordMapper(
+  function (acequire, exports) {
+    const oop = acequire('../lib/oop');
+    const { JavaScriptHighlightRules } = acequire(
+      './javascript_highlight_rules'
+    );
+
+    // MongoDB highligher is just javascript highligher with overriden
+    // identifier mapper
+    function MongoDBHighlightRules(...args) {
+      oop.mixin(this, new JavaScriptHighlightRules(...args));
+
+      const keywordMapper = this.createKeywordMapper(
         {
-          'variable.language':
-            'Code|ObjectId|Binary|BinData|UUID|DBRef|Timestamp|NumberInt|NumberLong|NumberDecimal|MaxKey|MinKey|ISODate|RegExp',
+          // BSON types are mapped as global vars
+          'variable.language.bson': toMapperVal(BSON_TYPES),
+
+          // Operator are mapped as functions (see README for context)
+          'support.function.accumulator': toMapperVal(ACCUMULATORS),
+          'support.function.expression_op': toMapperVal(EXPRESSION_OPERATORS),
+          'support.function.conversion_op': toMapperVal(CONVERSION_OPERATORS),
+          'support.function.query_op': toMapperVal(QUERY_OPERATORS),
+          'support.function.stage_op': toMapperVal(STAGE_OPERATORS),
```

Comment:
Adding a third token here (bson, accumulator, query_op, etc) doesn't change old behavior, but might be helpful later